charts/osm/templates/prometheus-configmap.yaml

```diff
@@ -111,6 +111,123 @@ data:
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: source_kind
+          replacement: $3
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: source_name
+          replacement: $4
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: source_pod
+          replacement: $5
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: destination_namespace
+          replacement: $6
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: destination_kind
+          replacement: $7
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: destination_name
+          replacement: $8
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_total
+          target_label: destination_pod
+          replacement: $9
+        - source_labels: [__name__]
+          action: replace
+          regex: .*osm_request_total
+          target_label: __name__
+          replacement: osm_request_total
+
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_source_namespace_(.*)_source_kind_(.*)_source_name_(.*)_source_pod_(.*)_destination_namespace_(.*)_destination_kind_(.*)_destination_name_(.*)_destination_pod_(.*)_osm_request_duration_ms_(bucket|sum|count)
```

Comment:
Wow. Is the relabeling necessary? I'd have expected the labeling should have worked out of the box with no relabeling needed. But now thinking about, it rings a bell of this issue.

If so:
`.*envoy_source_namespace_(.*)_source_kind_.*` would this work?
I think it will save quite some cycles if you avoided parsing all vars at each relabel.

At current conf, with some pods (400 or more), prometheus it reaches 0.6-0.8 cpu constant cpu usage just by parsing the envoys (probably also computing grafana requests since I'm looking). I was already thinking we will have to reduce the scraping interval at current rate; adding this will make this angle of matters worse.

We should see what can we remove from Line 57 also now that we have this. I honestly hope most of it.

Reply:
Yeah, it should work to only capture one group per regex. Originally I only did it this way because it was easier to copy-paste, didn't consider the performance aspect. I'll make that change.